storage/src/backend/registry.rs

```diff
@@ -487,8 +491,31 @@ impl RegistryReader {
                 return self._try_read(buf, offset, false);
             }
         } else {
-            resp =
-                self.request::<&[u8]>(Method::GET, url.as_str(), None, headers.clone(), false)?;
+            // resp =
+            //     self.request::<&[u8]>(Method::GET, url.as_str(), None, headers.clone(), false)?;
+            resp = match self.request::<&[u8]>(
+                Method::GET,
+                url.as_str(),
+                None,
+                headers.clone(),
+                false,
+            ) {
+                Ok(res) => res,
+                Err(e) => {
+                    if &url[0..5] != "https" {
+                        return Err(e);
+                    }
+                    if self.state.scheme.read().unwrap().clone() == "https" {
```

Comment:
We need to check if the error message matches `server gave HTTP response to HTTPS client` or `connect: connection refused` (not sure about the concrete message).

Reply:
When we send `https` to `http` server, we will get this error: 
```
reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Ipv4(10.26.200.20)), port: Some(5000), path: "/v2/my-ubuntu-nydus/blobs/sha256:6fabb2fb4a5c017e9de1fb1dc044b69aa96ea29b0928da6330f674d51ab16e51", query: None, fragment: None }, source: hyper::Error(Connect, Error { code: -9847, message: "record overflow" }) }
```

Reply:
It seems the error happens when creating the TLS connection, maybe we can reference the golang implementation here: https://github.com/golang/go/blob/f21be2fdc6f1becdbed1592ea0b245cdeedc5ac8/src/net/http/client.go#L262